src/test/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtAddressControllerTest.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.dts.fact.controllers.admin;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.dts.fact.exception.NotFoundException;
+import uk.gov.hmcts.dts.fact.model.admin.CourtAddress;
+import uk.gov.hmcts.dts.fact.services.admin.AdminCourtAddressService;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(AdminCourtAddressController.class)
+@AutoConfigureMockMvc(addFilters = false)
+public class AdminCourtAddressControllerTest {
+    private static final String TEST_SLUG = "court-slug";
+    private static final String BASE_PATH = "/admin/courts/";
+    private static final String ADDRESSES_PATH = "/" + "addresses";
+
+    private static final List<String> ADDRESS1 = Arrays.asList("first address line 1", "first address line 2");
+    private static final List<String> ADDRESS2 = Arrays.asList("second address line 1", "second address line 2");
+    private static final String TOWN_NAME1 = "first town";
+    private static final String TOWN_NAME2 = "second town";
+    private static final String POSTCODE1 = "first postcode";
+    private static final String POSTCODE2 = "second postcode";
+
+    private static final List<CourtAddress> COURT_ADDRESSES = Arrays.asList(
+        new CourtAddress(1, ADDRESS1, null, TOWN_NAME1, null, POSTCODE1),
+        new CourtAddress(2, ADDRESS2, null, TOWN_NAME2, null, POSTCODE2)
+    );
+    private static String courtAddressesJson;
+
+    @Autowired
+    private transient MockMvc mockMvc;
+
+    @MockBean
+    private AdminCourtAddressService adminService;
+
+    @BeforeAll
+    static void setUp() throws JsonProcessingException {
+        courtAddressesJson = new ObjectMapper().writeValueAsString(COURT_ADDRESSES);
+    }
+
+    @Test
+    void shouldRetrieveCourtAddresses() throws Exception {
+        when(adminService.getCourtAddressesBySlug(TEST_SLUG)).thenReturn(COURT_ADDRESSES);
+
+        mockMvc.perform(get(BASE_PATH + TEST_SLUG + ADDRESSES_PATH))
```

Comment:
Super minor again, string format if you want. Although I don't think we do it for the other tests, so dw if its a pain